theano/sandbox/mkl/mkl_type.py

```diff
@@ -0,0 +1,270 @@
+from __future__ import absolute_import, print_function, division
+import os
+import numpy
+import theano
+from theano import Type, Variable
+from theano import tensor
+from six import StringIO
+
+try:
+    import mkl_ndarray
+    import mkl_ndarray.mkl_ndarray as mkl
+
+    try:
+        mkl_ndarray.__file__
+    except AttributeError:
+        from theano.gof.cmodule import get_lib_extension
+        mkl_ndarray.__file__ = os.path.join(mkl_ndarray.__path__._path[0], 'mkl_ndarray.' + get_lib_extension())
+except ImportError:
+    mkl = None
+
+
+class _operators(tensor.basic._tensor_py_operators):
+    dtype = property(lambda s: s.type.dtype)
+    broadcastable = property(lambda s: s.type.broadcastable)
+    ndim = property(lambda s: s.type.ndim)
+
+
+class MKLNdarrayVariable(_operators, Variable):
+    pass
+
+
+class MKLNdarrayType(Type):
```

Comment:
I understand your last explanation. But I don't understand why this is a reason for a new C object. shapes and strides in the data object are enough to specify the memory layout of your point 1) 2 comments above.